packages/flutter/test/material/time_picker_test.dart

```diff
@@ -696,126 +607,166 @@ void _tests() {
+    await tester.enterText(find.byType(TextField).first, '8');
+    await tester.enterText(find.byType(TextField).last, '15');
+    await tester.tap(find.byIcon(Icons.access_time));
+    await finishPicker(tester);
+    expect(result, equals(const TimeOfDay(hour: 8, minute: 15)));
+  });
 
-  final WidgetTester tester;
-  final CustomPainter painter;
-  final SemanticsTester semantics;
-  final PaintPattern expectedLabels = paints;
-  final List<_SemanticsNodeExpectation> expectedNodes = <_SemanticsNodeExpectation>[];
+  testWidgets('Invalid text prevents dismissing', (WidgetTester tester) async {
+    TimeOfDay result;
+    await startPicker(tester, (TimeOfDay time) { result = time; }, entryMode: TimePickerEntryMode.input);
 
-  void addLabel(String label, double left, double top, double right, double bottom) {
-    expectedNodes.add(_SemanticsNodeExpectation(label, left, top, right, bottom));
-  }
+    // Invalid hour.
+    await tester.enterText(find.byType(TextField).first, '88');
+    await tester.enterText(find.byType(TextField).last, '15');
+    await finishPicker(tester);
+    expect(result, null);
 
-  void assertExpectations() {
-    final TestRecordingCanvas canvasRecording = TestRecordingCanvas();
-    painter.paint(canvasRecording, const Size(220.0, 220.0));
-    final List<ui.Paragraph> paragraphs = canvasRecording.invocations
-      .where((RecordedInvocation recordedInvocation) {
-        return recordedInvocation.invocation.memberName == #drawParagraph;
-      })
-      .map<ui.Paragraph>((RecordedInvocation recordedInvocation) {
-        return recordedInvocation.invocation.positionalArguments.first as ui.Paragraph;
-      })
-      .toList();
-
-    final PaintPattern expectedLabels = paints;
-    int i = 0;
-
-    for (final _SemanticsNodeExpectation expectation in expectedNodes) {
-      expect(semantics, includesNodeWith(value: expectation.label));
-      final Iterable<SemanticsNode> dialLabelNodes = semantics
-          .nodesWith(value: expectation.label)
-          .where((SemanticsNode node) => node.tags?.contains(const SemanticsTag('dial-label')) ?? false);
-      expect(dialLabelNodes, hasLength(1), reason: 'Expected exactly one label ${expectation.label}');
-      final Rect rect = Rect.fromLTRB(expectation.left, expectation.top, expectation.right, expectation.bottom);
-      expect(dialLabelNodes.single.rect, within(distance: 1.0, from: rect),
-        reason: 'This is checking the node rectangle for label ${expectation.label}');
-
-      final ui.Paragraph paragraph = paragraphs[i++];
-
-      // The label text paragraph and the semantics node share the same center,
-      // but have different sizes.
-      final Offset center = dialLabelNodes.single.rect.center;
-      final Offset topLeft = center.translate(
-        -paragraph.width / 2.0,
-        -paragraph.height / 2.0,
-      );
+    // Invalid hour.
+    await tester.enterText(find.byType(TextField).first, '8');
```

Comment:
Why is this an invalid hour?

Reply:
Oops comment should read invalid minute, fixed